### PR TITLE
[2.x] Laravel 10 support

### PR DESCRIPTION
**Pull request by antonioribeiro**
Just installed it locally, tests are passing.

### CODE REVIEW

**Comment by ifox** [comment]
Thanks @antonioribeiro!

Some of our dependencies are not supporting Laravel 10 yet, as noted on https://github.com/area17/twill/pull/2122#issuecomment-1415763778.

`lazychaser/nested-set` is an optional dependency, so not a blocker, but `cartalyst/tags` and `laravel-translatable` would prevent the install at the moment, no?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -32,10 +32,12 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        php: [8.1, 8.0, 7.4, 7.3]
-        laravel: [9.*, 8.*, 7.*, 6.*, 5.8.*]
+        php: [8.2, 8.1, 8.0, 7.4, 7.3]
+        laravel: [10.8, 9.*, 8.*, 7.*, 6.*, 5.8.*]
```
Comment (ifox) [inline .github/workflows/main.yml:36]: I guess you meant 10.* here

Comment (antonioribeiro) [reply .github/workflows/main.yml]: Yup, fixed, thanks!

**Comment by antonioribeiro** [comment]
Yeah, you are right, @ifox , closing this for now.

**Comment by ifox** [comment]
I think we can keep it open? Definitely want to release support for Laravel 10 on Twill 2.

**Comment by antonioribeiro** [comment]
Sure I'll reopen it as soon as we are able to close all subjects. I'm waiting on the nestedset guys to merge one of those PRs :D

**Comment by antonioribeiro** [comment]
@ifox , all PHPUnit tests are now passing, but we have some flags on `roave-backwards-compatibility-check` that must be thought before we move forward. This is one example:

```php
    protected $dates = [
        'deleted_at',
    ];
```

It was deprecated in favor of `$casts` on Laravel 8, and now removed from Laravel 10, so we are supposed to change the implementation but should we keep supporting Laravel versions that doesn't have $casts, are we still supporting any?

**Comment by ifox** [comment]
Thanks @antonioribeiro. 

We do indeed have to take a closer look at this change, because we also rely on `getDates` from Eloquent, which implementation has changed in Laravel 10: https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L1441. Before, it was using the `$dates` model property: https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L1450. And this is where we rely on it: https://github.com/area17/twill/blob/748467bbf9177e8d1e38086d1c959e2e6bc9f020/src/Repositories/Behaviors/HandleDates.php#L17.

We are not supporting any version that didn't have `$casts`, as it has been available since 5.0.

**Comment by antonioribeiro** [comment]
@ifox, I just [reimplemented](https://github.com/area17/twill/pull/2155/files#diff-9b43f5d7163463bfe69ad4642f5455421ba9b63ed4fbd5d096df29a1b64868a0L123) `getDates()` to support Laravel 10 getting from `$dates` and created some tests to check if it's all working fine.

Most of the other errors can probably be ignored, but I will still look at them all tomorrow

**Comment by antonioribeiro** [comment]
@ifox , here's the analysis of the breaking change errors we get:

### BROKEN - Needs a fix -- Please review the fix
```
Error: Property Illuminate\Database\Eloquent\Model#$dates was removed
```

### Can probably be ignored
```
Error: Default parameter value for parameter $inverse of Illuminate\Database\Eloquent\Concerns\HasRelationships#morphToMany() changed from false to NULL

Error: Parameter 7 of Illuminate\Database\Eloquent\Concerns\HasRelationships#morphToMany() changed name from inverse to relation
```

### Not being used directly (internal calls)
```
Error: Method Illuminate\Foundation\Bus\Dispatchable::dispatchNow() was removed

Error: Method Illuminate\Foundation\Bus\DispatchesJobs#dispatchNow() was removed

Error: Parameter 4 of Illuminate\Foundation\Validation\ValidatesRequests#validateWithBag() changed name from customAttributes to attributes

Error: The parameter $parameters of Illuminate\Support\Traits\Macroable#__call() changed from no type to a non-contravariant array

Error: The parameter $parameters of Illuminate\Support\Traits\Macroable#__call() changed from no type to array

Error: Parameter 3 of Illuminate\Foundation\Validation\ValidatesRequests#validate() changed name from customAttributes to attributes

```

**Pull request merged**